pkg/console/controllers/clidownloads/controller.go

```diff
@@ -0,0 +1,294 @@
+package clidownloads
+
+import (
+	// standard lib
+	"fmt"
+	"time"
+
+	// kube
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+
+	// openshift
+	v1 "github.com/openshift/api/console/v1"
+	operatorsv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourcemerge"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	// informers
+	consoleinformersv1 "github.com/openshift/client-go/console/informers/externalversions/console/v1"
+	operatorinformersv1 "github.com/openshift/client-go/operator/informers/externalversions/operator/v1"
+	routesinformersv1 "github.com/openshift/client-go/route/informers/externalversions/route/v1"
+
+	// listers
+	operatorlistersv1 "github.com/openshift/client-go/operator/listers/operator/v1"
+
+	// clients
+	consoleclientv1 "github.com/openshift/client-go/console/clientset/versioned/typed/console/v1"
+	routeclientv1 "github.com/openshift/client-go/route/clientset/versioned/typed/route/v1"
+
+	// operator
+	"github.com/openshift/console-operator/pkg/api"
+	"github.com/openshift/console-operator/pkg/console/status"
+	routesub "github.com/openshift/console-operator/pkg/console/subresource/route"
+	"github.com/openshift/console-operator/pkg/console/subresource/util"
+)
+
+const (
+	controllerWorkQueueKey = "clidownloads-sync-work-queue-key"
+	controllerName         = "ConsoleCLIDownloadsSyncController"
+
+	ocCLIDownloadsCustomResourceName  = "oc-cli-downloads"
+	odoCLIDownloadsCustomResourceName = "odo-cli-downloads"
+)
+
+type CLIDownloadsSyncController struct {
+	// clients
+	consoleCliDownloadsClient consoleclientv1.ConsoleCLIDownloadInterface
+	routeClient               routeclientv1.RoutesGetter
+	// listers
+	operatorConfigLister operatorlistersv1.ConsoleLister
+	// events
+	cachesToSync []cache.InformerSynced
+	queue        workqueue.RateLimitingInterface
+	recorder     events.Recorder
+}
+
+func NewCLIDownloadsSyncController(
+	// clients
+	operatorClient v1helpers.OperatorClient,
+	cliDownloadsInterface consoleclientv1.ConsoleCLIDownloadInterface,
+	routeClient routeclientv1.RoutesGetter,
+	// informers
+	operatorConfigInformer operatorinformersv1.ConsoleInformer,
+	consoleCLIDownloadsInformers consoleinformersv1.ConsoleCLIDownloadInformer,
+	routesInformers routesinformersv1.RouteInformer,
+	// recorder
+	recorder events.Recorder,
+) *CLIDownloadsSyncController {
+
+	ctrl := &CLIDownloadsSyncController{
+		// clients
+		consoleCliDownloadsClient: cliDownloadsInterface,
+		routeClient:               routeClient,
+		// listers
```

Comment:
You don't need the `operatorconfiglister` here.  We only need the `informer` below to `Inform()` & kick our watches.  And we only need a `client` for when we need to do a `.Get()`.

Reply:
After a little more research and a few more conversations, confirmed we only need the `Informer` and `client` for the actual `CLIDownloads` resource.  We shouldn't watch operator config or anything else.